### PR TITLE
DT-964: added user exclusion / restriction secure endpoint

### PR DESCRIPTION
**Pull request by SimonMitchellMOJ**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java [hunk 1]
```diff
@@ -181,22 +186,22 @@
...
-        value = {
-            @ApiResponse(code = 200, message = "OK", response = OffenderDocuments.class),
-            @ApiResponse(code = 400, message = "Invalid request", response = ErrorResponse.class),
-            @ApiResponse(code = 401, message = "Unauthorised", response = ErrorResponse.class),
-            @ApiResponse(code = 403, message = "Forbidden", response = ErrorResponse.class),
-            @ApiResponse(code = 404, message = "Not Found. For example if the CRN is not known.", response = ErrorResponse.class),
-            @ApiResponse(code = 500, message = "Unrecoverable error whilst processing request.", response = ErrorResponse.class)
-        })
+            value = {
+                    @ApiResponse(code = 200, message = "OK", response = OffenderDocuments.class),
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java:190]: in general shouldn't document the 200 `OK` response as will be auto generated so therefore more likely to be correct

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java [hunk 2]
```diff
@@ -486,12 +491,30 @@ public CustodialStatus getCustodialStatusBySentenceId(
...
+    public ResponseEntity<AccessLimitation> checkUserAccessByCrn(
+            final @PathVariable("crn") String crn) {
+        Optional<OffenderDetail> maybeOffender = offenderService.getOffenderByCrn(crn);
+
+        return maybeOffender.isEmpty() ? new ResponseEntity<>(NOT_FOUND) : accessLimitationResponseEntityOf(maybeOffender.get());
+    }
+
+    private ResponseEntity<AccessLimitation> accessLimitationResponseEntityOf(OffenderDetail offender) {
+
+        AccessLimitation accessLimitation = userService.accessLimitationOf(currentUserSupplier.username().get(), offender);
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java:516]: variables can be `final`

File: src/test/java/uk/gov/justice/digital/delius/controller/secure/OffenderResource_CheckUserAccess.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+                .as(AccessLimitation.class);
+
+        assertThat(accessLimitation.isUserExcluded()).isTrue();
+        assertThat(accessLimitation.getExclusionMessage()).isEqualTo("Your excluded");
+        assertThat(accessLimitation.isUserRestricted()).isTrue();
+        assertThat(accessLimitation.getRestrictionMessage()).isEqualTo("Your Restricted");
+
+
+    }    @Test
+    void willCheckUserAccessOffenderNotFound(){
```
Comment (mikehalmamoj) [inline src/test/java/uk/gov/justice/digital/delius/controller/secure/OffenderResource_CheckUserAccess.java:100]: The swagger doc mentioned a not found for unknown user too - does that need a test?

Comment (mikehalmamoj) [reply src/test/java/uk/gov/justice/digital/delius/controller/secure/OffenderResource_CheckUserAccess.java:100]: Sorry ignore me - just found that one in the integration test :smile:

Comment (SimonMitchellMOJ) [reply src/test/java/uk/gov/justice/digital/delius/controller/secure/OffenderResource_CheckUserAccess.java:100]: Ignored !

**Pull request merged**